src/Bicep.Core/FileSystem/PathHelper.cs

```diff
@@ -23,14 +24,21 @@ public static class PathHelper
         /// </summary>
         /// <param name="path">The path.</param>
         /// <param name="baseDirectory">The base directory to use when resolving relative paths. Set to null to use CWD.</param>
-        public static string ResolvePath(string path, string? baseDirectory = null)
+        /// <param name="fileSystem">The file system abstraction.</param>
+        public static string ResolvePath(string path, string? baseDirectory = null, IFileSystem? fileSystem = null)
```

Comment:
Again with the nulls.  Why not:

  fileSystem ??= new IOFileSystem();
  
  Then you don't need to do all the if/thens.

Reply:
Makes sense - I had nearly overlooked that class. I've updated the code accordingly.